test/compile.jl

```diff
@@ -71,6 +72,16 @@ try
     end
     Base.compilecache("FooBar")
     sleep(2)
+    @test isfile(joinpath(dir, "FooBar.ji"))
+
+    touch(FooBar_file)
+    insert!(Base.LOAD_CACHE_PATH, 1, dir2)
+    Base.recompile_stale(:FooBar, joinpath(dir, "FooBar.ji"))
+    sleep(2)
```

Comment:
This still isn't very future-proof. Needs improvement before I'd be comfortable backporting this. We shouldn't be making assumptions about how long distro buildbots, limited-resource VM's, or small embedded processors are going to take to do codegen here.